substrate/frame/revive/src/evm/api/rlp_codec.rs

```diff
@@ -96,6 +96,31 @@ impl TransactionSigned {
 		s.out().to_vec()
 	}
 
+	/// Encode the Ethereum transaction type into bytes.
+	///
+	/// This is needed to encode the receipts.
+	pub fn signed_type(&self) -> Vec<u8> {
+		let mut s = rlp::RlpStream::new();
+
+		match &self {
+			TransactionSigned::Transaction4844Signed(tx) => {
+				s.append(&tx.transaction_4844_unsigned.r#type.value());
+			},
+			TransactionSigned::Transaction1559Signed(tx) => {
+				s.append(&tx.transaction_1559_unsigned.r#type.value());
+			},
+			TransactionSigned::Transaction2930Signed(tx) => {
+				s.append(&tx.transaction_2930_unsigned.r#type.value());
+			},
+			TransactionSigned::Transaction7702Signed(tx) => {
+				s.append(&tx.transaction_7702_unsigned.r#type.value());
+			},
+			TransactionSigned::TransactionLegacySigned(_) => {},
+		};
+
+		s.out().to_vec()
+	}
+
```

Comment:
seems to be unused , also here you can just return the type directly instead of creating a rlp::RlpStream